src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php

```diff
@@ -1025,6 +1027,42 @@ protected function createChildContext(array $parentContext, string $attribute, ?
 
         $this->assertFalse($normalizer->childContextCacheKey);
     }
+
+    public function testNormalizationWithMaxDepthOnStdclassObjectDoesNotThrowWarning()
+    {
+        $object = new \stdClass();
+        $object->string = 'yes';
+
+        $loader = new LoaderChain([
+            new AttributeLoader(),
+        ]);
+        $classMetadataFactory = new ClassMetadataFactory($loader);
+
+        $serializer = new Serializer([
+            new ObjectNormalizer($classMetadataFactory),
+        ], [
+            new JsonEncoder(),
+        ]);
```

Comment:
```suggestion
        $normalizer = new ObjectNormalizer($classMetadataFactory);
```
You're only interacting with the normalization stage, therefore you can use directly the normalizer

Reply:
thanks, reduced to just calling the normalizer